addons/binding/org.openhab.binding.regoheatpump/src/main/java/org/openhab/binding/regoheatpump/handler/Rego6xxHeatPumpHandler.java

```diff
@@ -0,0 +1,405 @@
+    }
+
+    private void readLastErrorTimestamp() {
+        readLastError(CHANNEL_LAST_ERROR_TIMESTAMP, e -> new DateTimeType(e.timestamp()));
+    }
+
+    private void readLastError(final String channelIID, final Function<ErrorLine, State> converter) {
+        executeCommandAndUpdateState(channelIID, CommandFactory.createReadLastErrorCommand(),
+                ResponseParserFactory.ErrorLine, e -> {
+                    return e == null ? UnDefType.NULL : converter.apply(e);
+                });
+    }
+
+    private void readFromFrontPanel(final String channelIID, short address) {
+        final byte[] command = CommandFactory.createReadFromFrontPanelCommand(address);
+        executeCommandAndUpdateState(channelIID, command, ResponseParserFactory.Short, DecimalType::new);
+    }
+
+    private void readFromSystemRegister(final String channelIID) {
+        final RegoRegisterMapper.Channel channel = mapper.map(channelIID);
+        if (channel != null) {
+            final byte[] command = CommandFactory.createReadFromSystemRegisterCommand(channel.address());
+            executeCommandAndUpdateState(channelIID, command, ResponseParserFactory.Short, channel::convert);
+        } else {
+            logger.debug("Unsupported channel {}", channelIID);
+        }
+    }
+
+    private synchronized <T> void executeCommandAndUpdateState(final String channelIID, final byte[] command,
+            final ResponseParser<T> parser, Function<T, State> converter) {
+
+        if (logger.isDebugEnabled()) {
+            logger.debug("Reading value for channel '{}' ...", channelIID);
+        }
+
+        try {
+
+            final T result = executeCommandWithRetry(channelIID, command, parser, 5);
+
+            if (logger.isDebugEnabled()) {
+                logger.debug("Got value for '{}' = {}", channelIID, result);
+            }
+
+            updateState(channelIID, converter.apply(result));
+
+        } catch (IOException e) {
+
+            logger.warn("Accessing value for channel '" + channelIID + "' failed.", e);
+
+            synchronized (channelDescriptors) {
+                channelDescriptors.clear();
+            }
+
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+
+        } catch (InterruptedException e) {
+
+            logger.debug("Execution interrupted when accessing value for channel '" + channelIID + "'", e);
+            Thread.currentThread().interrupt();
+
+        } catch (Exception e) {
+
+            logger.warn("Accessing value for channel '" + channelIID + "' failed.", e);
+        }
+    }
+
+    private <T> T executeCommandWithRetry(final String channelIID, final byte[] command, final ResponseParser<T> parser,
+            int retry) throws InterruptedException, IOException {
+
+        try {
+            checkRegoDevice();
+            return executeCommand(channelIID, command, parser);
+
+        } catch (Exception e) {
+
+            logger.debug(
```

Comment:
You should either log an exception or let it pass, maybe you should move this debug statement in the `if` on line 267